src/gui/editorwidgets/qgsjsoneditwidgetfactory.cpp

```diff
@@ -55,3 +55,8 @@ unsigned int QgsJsonEditWidgetFactory::fieldScore( const QgsVectorLayer *vl, int
       break;
   }
 }
+
+bool QgsJsonEditWidgetFactory::isReadOnly() const
+{
+  return true;
```

Comment:
Hmm - looking into this now, the json editor IS editable if it's not set to the "Tree" view mode. And users can switch that view at any time... so this one SHOULD return false, and remain non-read only.

Which means #47755 was invalid in the first place :upside_down_face: 

At least this fix still applies to the uuid/hidden wrappers!

Reply:
Actually ignore that... even in non-tree mode I see it's still read only. I'm not sure why that is!